server/src/jobs/etablissements/tags.js

```diff
@@ -0,0 +1,54 @@
+const { runScript } = require("../scriptWrapper");
+const logger = require("../../common/logger");
+const { Etablissement, Formation } = require("../../common/model");
+
+const etablissementTags = async () => {
+  logger.info("Settings tags on 'etablissements'...");
+
+  const cursor = Etablissement.find({}).cursor();
+  let count = 0,
+    updated = 0,
+    deleted = 0;
+  for await (const etablissement of cursor) {
+    (count++ + 1) % 100 === 0 && console.log(count);
+    try {
+      const formations = await Formation.find({
+        published: true,
+        $or: [
+          { etablissement_gestionnaire_id: etablissement._id },
+          { etablissement_formateur_id: etablissement._id },
+          { etablissement_responsable_id: etablissement._id },
+        ],
+      }).select("tags");
+
+      const tags = [...new Set(formations.flatMap((formation) => formation.tags))].sort();
```

Comment:
Tu peux aussi passer directement par un `distinct` pour récupérer les tags :)
```suggestion
      const tags = await Formation.distinct("tags", {
        published: true,
        $or: [
          { etablissement_gestionnaire_id: etablissement._id },
          { etablissement_formateur_id: etablissement._id },
          { etablissement_responsable_id: etablissement._id },
        ],
      });
```

Reply:
Pas sûr que ça me sorte une array unique, étant donné que tags est lui-même une array. il faudra tout de même utiliser  le flatMap et le [...new Set()]. Mais ça peut remplacer la première partie

Reply:
Corrigé